scipy/stats/tests/test_multicomp.py

```diff
@@ -250,6 +250,89 @@ def test_alternatives(self, alternative, statistic, pvalue):
             )
             assert_allclose(res.pvalue, pvalue, atol=1e-7)
 
+    # filter warning that could happen during optimization
+    @pytest.mark.parametrize(
+        'alternative, allowance, ci_low, ci_high',
+        [
+            (
+                'two-sided', 11,
+                [0.7529028025053, -8.2470971974947, -15.2470971974947],
+                [21.2470971974947, 12.2470971974947, 5.2470971974947]
+             ),
+            (
+                'less', 9,
+                [2.4023682323149, -6.5976317676851, -13.5976317676851],
+                [np.inf, np.inf, np.inf]
+            ),
+            (
+                'greater', 9,
+                [-np.inf, -np.inf, -np.inf],
+                [19.5984402363662, 10.5984402363662, 3.5984402363662]
+            )
+        ]
+    )
+    def test_allowance(self, alternative, allowance, ci_low, ci_high):
+        # Example (a) from Dunnett1995 using R multcomp `glht`
+        # Same as above with confint(test) and the corresponding alternative
+        rng = np.random.default_rng(189117774084579816190295271136455278291)
+        samples = [
+            [55, 64, 64],
+            [55, 49, 52],
+            [50, 44, 41]
+        ]
+        control = [55, 47, 48]
+
+        res = stats.dunnett(
+            *samples, control=control, alternative=alternative,
+            random_state=rng
+        )
+        allowance_ = res._allowance(confidence_level=0.95)
+        assert_allclose(allowance_, allowance, atol=1)
```

Comment:
Does `glht` state the allowance? If not, how do you calculate it from what it provides?

Reply:
Not that I know. The link with the allowance is immediate though with the CI. It's just the mean of the samples +- the allowance. The few values we have here are from the literature, hence the low precision. But then we have a better precision for the CI. I think this is reasonable, even more because the objective is the CI which matches the example.